.github/workflows/tests.yml

```diff
@@ -62,13 +61,20 @@ jobs:
         python-version: ${{ matrix.python }}
         cache: pip
 
-    - name: Install and upgrade tox
+    - name: Install and upgrade test tools
       run: python -m pip install --progress-bar off --upgrade tox
 
+    - name: Cache
+      uses: actions/cache@v4
+      with:
+        path: |
+          .tox
+        key: ${{ matrix.toxenv }}-${{ runner.os }}-${{ hashFiles('requirements.txt', 'pyproject.toml') }}
```

Comment:
By creating a hash from these two files and including it in the cache name, we can make it so that the cache gets regenerated whenever `requirements.txt` or `pyproject.toml` change.